src/pages/dashboard/analysis/components/Charts/TagCloud/index.tsx

```diff
@@ -0,0 +1,212 @@
+import { Chart, Coord, Geom, Shape, Tooltip } from 'bizcharts';
+import React, { Component } from 'react';
+
+import DataSet from '@antv/data-set';
+import Debounce from 'lodash.debounce';
+import classNames from 'classnames';
+import autoHeight from '../autoHeight';
+import styles from './index.less';
+
+/* eslint no-underscore-dangle: 0 */
+/* eslint no-param-reassign: 0 */
+
+const imgUrl = 'https://gw.alipayobjects.com/zos/rmsportal/gWyeGLCdFFRavBGIDzWk.png';
+
+export interface TagCloudProps {
+  data: {
+    name: string;
+    value: number;
+  }[];
+  height?: number;
+  className?: string;
+  style?: React.CSSProperties;
+}
+
+interface TagCloudState {
+  dv: any;
+  height?: number;
+  width: number;
+}
+
+class TagCloud extends Component<TagCloudProps, TagCloudState> {
+  state = {
+    dv: null,
+    height: 0,
+    width: 0,
+  };
+
+  isUnmount: boolean = false;
+
+  requestRef: number = 0;
+
+  root: HTMLDivElement | undefined = undefined;
+
+  imageMask: HTMLImageElement | undefined = undefined;
+
+  componentDidMount() {
+    requestAnimationFrame(() => {
+      this.initTagCloud();
+      this.renderChart(this.props);
+    });
+    window.addEventListener('resize', this.resize, { passive: true });
+  }
+
+  componentDidUpdate(preProps?: TagCloudProps) {
+    const { data } = this.props;
+    if (preProps && JSON.stringify(preProps.data) !== JSON.stringify(data)) {
+      this.renderChart(this.props);
+    }
+  }
+
+  componentWillUnmount() {
+    this.isUnmount = true;
+    window.cancelAnimationFrame(this.requestRef);
+    window.removeEventListener('resize', this.resize);
+  }
+
+  resize = () => {
+    this.requestRef = requestAnimationFrame(() => {
+      this.renderChart(this.props);
+    });
+  };
+
+  saveRootRef = (node: HTMLDivElement) => {
+    this.root = node;
+  };
+
+  initTagCloud = () => {
+    function getTextAttrs(cfg: {
+      x?: any;
+      y?: any;
+      style?: any;
+      opacity?: any;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=4583897)